components/clp-py-utils/clp_py_utils/profiling_utils.py

```
@@ -0,0 +1,256 @@
+    Saves profiler output to HTML and text formats. Generates .html and .txt files.
+
+    :param profiler: The pyinstrument Profiler object containing profiling data.
+    :param section_name: Name identifying this profiling section.
+    :param job_id: Optional job identifier for filename.
+    :param task_id: Optional task identifier for filename.
+    """
+    try:
+        # Get the session for logging
+        session = profiler.last_session
+        if not session:
+            logger.debug(f"No profiling session for {section_name}")
```

Ruff (0.14.1)

Logging statement uses f-string

(G004)